File: 7kyu/Shortest Word.js
// Simple, given a string of words, return the length of the shortest word(s).

// String will never be empty and you do not need to account for different data types.

//First Pass

function findShort(str){
    const arr = str.split(" ")
    
    const wordLengthArr = arr.map(word=> word.length)
  
    return Math.min(...wordLengthArr)
  }

//Code wars refractor recomendation

function findShort(s){
    return Math.min(...s.split(" ").map (s => s.length));
}

File: 7kyu/Shortest Word.py
# Simple, given a string of words, return the length of the shortest word(s).

# String will never be empty and you do not need to account for different data types.


# My Python Solution 

def find_short(s):
    str_word_list_length = map(len,s.split())
    
    return min(str_word_list_length)

# Code Wars Alternatives

def find_short(s):
    return min(len(x) for x in s.split())

# Or

def find_short(s):
    return len(min(s.split(' '), key=len))